deployment-detector/src/agent.spec.ts

```diff
@@ -24,54 +30,125 @@ const createMockArg = (agentId:number, owner:string, metaData:string, chainId:nu
   };
 };
 
-const arg = createMockArg(1,createAddress("0x01"),"QmPkydGrmSK2roUJeNzsdC3e7Yetr7zb7UNdmiXyRUM6ij", [1,137])
-const arg2= createMockArg(1,createAddress("0x01"),"QmPkydGrmSK2roUJeNzsdC3e7Yetr7zb7UNdmiXyRUM6ij", [1,80001])
+const mockarg = createMockArg(1, createAddress("0x05"), "QmPkydGrmSK2roUJeNzsdC3e7Yetr7zb7UNdmiXyRUM6ij", [137]);
 
+const mockarg2 = createMockArg(2, createAddress("0x06"), "QmPkydGrmSK2roUJeNzsdC3e7Yetr7zb7UNdmiXyRUM6if", [137]);
 
-describe("New bot deployment", () =>{
-  let handleTransaction: HandleTransaction
+describe("Nethermind deployer address bot test suite", () => {
+  let handleTransaction: HandleTransaction;
 
-  beforeAll(() =>{
-    handleTransaction = provideTransactionHandler(botsParams)
-  })
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(mockBotParams);
+  });
 
-  // Test-Case-One
+  it("ignores empty transactions", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent();
+    const findings: Finding[] = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
 
-  it("ignore empty transactions", async() =>{
-    const txEvent:TransactionEvent = new TestTransactionEvent()
-    const findings:Finding[] = await handleTransaction(txEvent)
-    expect(findings).toStrictEqual([])
-  })
+  it("ignores valid transactions (create bot) from a different address", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(otherDeployerAddress)
+      .setTo(mockFortaAgentRegistry)
+      .addEventLog(FORTA_AGENT_EVENT, mockOtherFortaContract, [
```

Comment:
```suggestion
      .setTo(mockFortaAgentRegistry)
      .addEventLog(FORTA_AGENT_EVENT, mockFortaAgentRegistry, [
```

Call should be to the correct Forta address - right now it should fail, because of the `txEvent.from` check missing in the logic.